src/ioctls/vm.rs

```diff
@@ -844,6 +890,42 @@ mod tests {
         assert!(vm_fd.register_irqfd(evtfd3, 5).is_err());
     }
 
+    #[test]
+    #[cfg(any(
+        target_arch = "x86",
+        target_arch = "x86_64",
+        target_arch = "arm",
+        target_arch = "aarch64"
+    ))]
+    fn test_unregister_irqfd() {
+        let kvm = Kvm::new().unwrap();
+        let vm_fd = kvm.create_vm().unwrap();
+        let evtfd1 = unsafe { eventfd(0, EFD_NONBLOCK) };
+        let evtfd2 = unsafe { eventfd(0, EFD_NONBLOCK) };
+        let evtfd3 = unsafe { eventfd(0, EFD_NONBLOCK) };
+        if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
+            assert!(vm_fd.register_irqfd(evtfd1, 4).is_ok());
+            assert!(vm_fd.register_irqfd(evtfd2, 8).is_ok());
+            assert!(vm_fd.register_irqfd(evtfd3, 4).is_ok());
+            assert!(vm_fd.unregister_irqfd(evtfd2, 8).is_ok());
+            // KVM irqfd doesn't report failure on this case:(
+            assert!(vm_fd.unregister_irqfd(evtfd2, 8).is_ok());
+        }
+
+        // On aarch64, this fails because setting up the interrupt controller is mandatory before
+        // registering any IRQ.
+        // On x86_64 this fails as the event fd was already matched with a GSI.
+        assert!(vm_fd.register_irqfd(evtfd3, 4).is_err());
+        assert!(vm_fd.register_irqfd(evtfd3, 5).is_err());
+
+        // KVM irqfd doesn't report failure on this case:(
+        assert!(vm_fd.unregister_irqfd(evtfd3, 5).is_ok());
+
+        let evtfd4 = unsafe { eventfd(0, EFD_NONBLOCK) };
+        unsafe { libc::close(evtfd4) };
+        assert!(vm_fd.unregister_irqfd(evtfd4, 5).is_err());
```

Comment:
This test is racy.
Might not occur now, but as we add more tests that deal with `fd`s we will hit the case where another test reuses the `fd` closed in the line above and then this check will sporadically fail.

I suggest you remove this part of the test, or test it with an always invalid `fd` like `-1`.

Reply:
Nice catch, thanks.

Reply:
Thanks for review:)